### PR TITLE
Update identifers and references in rule section of documentation

### PR DESCRIPTION
**Pull request by redhatrises**
Adds information about identifiers and references as well as mapping of identifiers and references.

### CODE REVIEW

File: docs/manual/developer_guide.adoc [hunk 1]
```diff
@@ -792,12 +792,103 @@ The <tt>/var/tmp</tt> directory is a world-writable directory used for temporary
...
+|DISA Control Correlation Identifiers (catalog identifier)
+|0-to-many
+|18,172,1403
+
+|srg, vmmsrg, etc.
+|DISA Security Requirements Guide (catalog identifier)
+|0-to-many
+|SRG-OS-000003-GPOS-00004
+
+|stigid@<product><produc_version>
```
Comment (jan-cerny) [inline docs/manual/developer_guide.adoc]: Missing t.

File: docs/manual/developer_guide.adoc [hunk 2]
```diff
@@ -792,12 +792,103 @@ The <tt>/var/tmp</tt> directory is a world-writable directory used for temporary
...
+|http://www.us-cert.gov/cas/techalerts
+|No
+|US-CERT technical cyber security alerts –the identifier value SHOULD be a technical cyber security alert ID (e.g., “TA05-189A”)
+|===
++
+When the rule is related to RHEL, it should have a CCE. A CEE (e.g. cce@rhel7: 80328-8) is used as a global identifier that maps the rule to the product over the lifetime of a rule. There should only be one CCE mapped to a rule as a global identifier. Any other usage of CCE is no longer considered a best practice.
 Available CCEs that can be assigned to new rules are listed in the `shared/references/cce-rhel-avail.txt` file.
-See `linux_os/guide/system/software/disk_partitioning/encrypt_partitions.rule` for an example of reference-type attributes.
++
+* `references`: This is related to the compliance document line items that the rule applies to. These can be attributes such as `stig`, `srg`, `nist`, etc., whose keys may be modified with a product (e.g., `stig@rhel6`) to restrict what products a reference identifier applies to. Depending on the type of reference (e.g. catalog, rule) will depend on how many can be added to a single rule.
```
Comment (matejak) [inline docs/manual/developer_guide.adoc]: STIG IDs (as well as CCEs) now have to be product-qualified, as we don't want Ubuntu STIG IDs.

File: docs/manual/developer_guide.adoc [hunk 3]
```diff
@@ -792,12 +792,103 @@ The <tt>/var/tmp</tt> directory is a world-writable directory used for temporary
...
+
+|pcidss
+|Payment Card Industry Data Security Standard
+|0-to-many, 0-to-1 is preferred
+|Req-8.7.c
+
+|And many others
+|===
++
+See `linux_os/guide/system/software/disk_partitioning/encrypt_partitions.rule` for an example of reference-type attributes as there are others that are not referenced above.
```
Comment (ggbecker) [inline docs/manual/developer_guide.adoc]:
```suggestion
See `linux_os/guide/system/software/disk_partitioning/encrypt_partitions/rule.yml` for an example of reference-type attributes as there are others that are not referenced above.
```

**Comment by jan-cerny** [comment]
@redhatrises Could you please take a look on the reviewer's comments? They seem easy to resolve.

File: docs/manual/developer_guide.adoc (hunk 1 above)
Comment (redhatrises) [reply docs/manual/developer_guide.adoc]: Fixed.

File: docs/manual/developer_guide.adoc (hunk 3 above)
Comment (redhatrises) [reply docs/manual/developer_guide.adoc]: Fixing it here, but really the entire doc is out-of-date for rule referencing and needs to be updated but is really out of the scope of this PR.

File: docs/manual/developer_guide.adoc (hunk 2 above)
Comment (redhatrises) [reply docs/manual/developer_guide.adoc]: Not 100% sure of what you are asking here, but I add a blurb that CCEs and STIGIDs apply to certain products and product versions and cannot be applied to multiple product/versions.

**Pull request merged**